common/src/main/java/sh/okx/civmodern/common/gui/screen/RadarConfigScreen.java

```diff
@@ -1,351 +1,441 @@
+                @Override
+                public @NotNull Component getText(final double value) {
+                    return Component.translatable("civmodern.screen.radar.x", String.valueOf((int) value));
+                }
             }
-
-            @Override
-            public Component getText(double value) {
-                return Component.translatable("civmodern.screen.radar.x", String.valueOf((int) value));
-            }
-        }));
-        addRenderableWidget(new DoubleOptionUpdateableSliderWidget(right, offset, 150, 20, 0, 300, 1, new DoubleValue() {
-
-            @Override
-            public double get() {
-                return config.getY();
+        ));
+        addRenderableWidget(new DoubleOptionUpdateableSliderWidget(
+            rightSideX,
+            offsetY,
+            Button.DEFAULT_WIDTH,
+            Button.DEFAULT_HEIGHT,
+            0, 300, 1,
+            new DoubleValue() {
+                @Override
+                public double get() {
+                    return RadarConfigScreen.this.config.getY();
+                }
+                @Override
+                public void set(final double value) {
+                    RadarConfigScreen.this.config.setY((int) value);
+                }
+                @Override
+                public @NotNull Component getText(final double value) {
+                    return Component.translatable("civmodern.screen.radar.y", String.valueOf((int) value));
+                }
             }
-
-            @Override
-            public void set(double value) {
-                config.setY((int) value);
-            }
-
-            @Override
-            public Component getText(double value) {
-                return Component.translatable("civmodern.screen.radar.y", String.valueOf((int) value));
-            }
-        }));
-        offset += 24;
-
-        foregroundColourY = backgroundColourY = offset;
-
-        offset += 12;
-
-        offset += 36;
-        addRenderableWidget(Button.builder(CommonComponents.GUI_DONE, button -> {
-            config.save();
-            Minecraft.getInstance().setScreen(parent);
-        }).pos(centre, offset).size(150, 20).build());
+        ));
+        offsetY += Button.DEFAULT_HEIGHT + 2;
+
+        this.bgPicker = addColourPicker(
+            leftSideX,
+            offsetY,
+            CivMapConfig.DEFAULT_RADAR_BG_COLOUR,
+            Component.literal("Background colour"),
+            this.config::getRadarBgColour,
+            this.config::setRadarBgColour,
+            this.colourProvider::setTemporaryRadarBackgroundColour
+        );
+        this.fgPicker = addColourPicker(
+            rightSideX,
+            offsetY,
+            CivMapConfig.DEFAULT_RADAR_FG_COLOUR,
+            Component.literal("Line colour"),
+            this.config::getRadarColour,
+            this.config::setRadarColour,
+            this.colourProvider::setTemporaryRadarForegroundColour
+        );
+        offsetY += 30 + 2;
```

Comment:
The colour pickers need to be added as the first thing because they can display on top of the buttons and should be handled first.